opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -416,6 +402,8 @@ def start_span(
         name: str,
         parent: ParentSpan = CURRENT_SPAN,
         kind: SpanKind = SpanKind.INTERNAL,
+        attributes: typing.Optional[types.Attributes] = None,
+        links: typing.Sequence[Link] = (),
```

Comment:
Do we intend to support iterator (delayed expansion)? This might give perf gain if the span is not sampled (so we don't have to evaluate the iterator at all).

Reply:
I think someone should read up on the motivation behind *lazy* links. I think lazily evaluated iterables may be a good fit for Python.

Reply:
This is a good point. If we're not using `__getitem__` -- and especially if we might not read the sequence at all, -- we should generally prefer `Iterator` over `Sequence`.